src/app/Dashboard/Diagnostics/DiagnosticsCard.tsx

```diff
@@ -0,0 +1,204 @@
+  duration: number;
+  period: number;
+}
+
+export enum DiagnosticsCardKind {}
+
+export function kindToId(kind: string): number {
+  return DiagnosticsCardKind[kind];
+}
+
+export const DiagnosticsCard: DashboardCardFC<DiagnosticsCardProps> = (props) => {
+  const { t } = useTranslation();
+  const serviceContext = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [running, setRunning] = React.useState(false);
+  const isError = React.useMemo(() => errorMessage != '', [errorMessage]);
+
+  const handleError = React.useCallback(
+    (error) => {
+      setErrorMessage(error.message);
+    },
+    [setErrorMessage],
+  );
+
+  const handleGC = React.useCallback(() => {
+    addSubscription(
+      serviceContext.api.runGC().subscribe({
+        error: (err) => handleError(err),
+      }),
+    );
+  }, [addSubscription, serviceContext.api, handleError]);
+
+  const GCButton = React.useMemo(() => {
+    return (
+      <Button
+        key={0}
+        aria-label={t('DaignosticsCard.DIAGNOSTICS_GC_BUTTON', { chartKind: props.chartKind })}
+        onClick={handleGC}
+        variant="plain"
+        icon={<SyncAltIcon />}
+        isDisabled={false}
+      />
+    );
+  }, [t, props.chartKind, handleGC]);
+
+  const actions = React.useMemo(() => {
+    const a = props.actions || [];
+    return [GCButton, ...a];
+  }, [props.actions, GCButton]);
+
+  const gcButtonLoadingProps = React.useMemo(
+    () =>
+      ({
+        spinnerAriaValueText: 'Invoke GC',
+        spinnerAriaLabel: 'saving-credentials',
+        isLoading: running,
+      }) as LoadingProps,
+    [running],
+  );
+
+  const header = React.useMemo(() => {
+    return (
+      <CardHeader
+        actions={{
+          actions: <>{actions}</>,
+          hasNoOffset: false,
+          className: undefined,
+        }}
+      >
+        <CardTitle>
+          {t('DiagnosticsCard.DIAGNOSTICS_CARD_TITLE', {
+            chartKind: props.chartKind,
+            duration: props.duration,
+            period: props.period,
+          })}
+        </CardTitle>
+      </CardHeader>
+    );
+  }, [props.chartKind, props.duration, props.period, t, actions]);
+
+  return isError ? (
+    <ErrorView title={'Error executing diagnostic command'} message={`${errorMessage}`} />
+  ) : (
+    <DashboardCard
+      id={props.chartKind + '-chart-card'}
+      dashboardId={props.dashboardId}
+      cardSizes={DiagnosticsCardSizes}
+      isCompact
+      cardHeader={header}
+      title={props.chartKind}
+      isDraggable={props.isDraggable}
+      isResizable={props.isResizable}
+      isFullHeight={props.isFullHeight}
+    >
+      <CardBody>
+        <Bullseye>
+          <EmptyState variant={EmptyStateVariant.lg}>
+            <EmptyStateHeader
+              titleText={<>{t('DiagnosticsCard.DIAGNOSTICS_CARD_TITLE')}</>}
+              icon={<EmptyStateIcon icon={DataSourceIcon} />}
```

Comment:
Same here about the DataSourceIcon - I think this should match the icon that represents this card in the card catalog, ex. the WrenchIcon perhaps?